interfaces/cuda/Control.cu

```diff
@@ -147,6 +147,13 @@ void ConcreteAPI::finalize() {
   if (status[StatusID::InterfaceInitialized]) {
     cudaStreamDestroy(defaultStream); CHECK_ERR;
     cudaEventDestroy(defaultStreamEvent); CHECK_ERR;
+    if (not genericStreams.empty()) {
```

Comment:
I still believe that not is a very strange coding style :(

Reply:
reworked. Now, it is `!genericStreams.empty()`

PS I don't know. It looks more informative/readable for me. The only one thing to keep in mind that `not` works only with `bool`. But `empty()` always returns bool. However, I just changed as you requested.